src/sentry/rules/processing/delayed_processing.py

```diff
@@ -159,7 +161,17 @@ def get_rules_to_fire(
                     ):
                         for group_id in rules_to_groups[alert_rule.id]:
                             if results[group_id] > target_value:
+                                if action_match == "all":
```

Comment:
I can't comment on the line, but rather than iterating through `condition_group_results`, why don't we do
```
unique_condition = UniqueCondition(condition_id, condition_interval, alert_rule.environment_id)
results = condition_group_results.get(unique_condition)
if results:
    ...
```